src/Java.Interop.Tools.Maven/Models/Project.Partial.cs

```diff
@@ -0,0 +1,62 @@
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Xml;
+using System.Xml.Serialization;
+using Java.Interop.Tools.Maven.Extensions;
+
+namespace Java.Interop.Tools.Maven.Models;
+
+public partial class Project
+{
+	public static Project Parse (Stream stream)
```

Comment:
Given that `Stream` will *also* contain XML, why have a `ParseXml()` method vs naming both `(Stream)` and `(string)` overloads `Parse()`?

Relatedly, there should probably *also* be a `Parse(XmlTextReader)` overload, so that both `Parse(Stream)` and `Parse(string)` can share most of their logic.

Reply:
*Alternate* naming logic: do what `XDocument` does, which has:

  * [`XDocument.Load()`](https://learn.microsoft.com/en-us/dotnet/api/system.xml.linq.xdocument.load?view=net-8.0) loads from `Stream`s, `XmlReader`s, etc.  `Load(string)` loads from a *file*.
  * [`XDocument.Parse()`](https://learn.microsoft.com/en-us/dotnet/api/system.xml.linq.xdocument.parse?view=net-8.0) loads from XML.

Thus, I think we should rename `Parse(Stream)` to `Load(Stream)`, add a `Load(XmlReader)` overload, and rename `ParseXml(string)` to `Parse(string)`:

```csharp
public static Project Load(XmlReader reader)
{
    var serializer = new XmlSerializer (typeof (Project));
    return (Project) serializer.Deserialize (reader);
}

public static Project Load(Stream stream)
{
    return Load (new XmlTextReader (stream) {
        Namespaces = false;
    });
}

public static Project Parse(string xml)
{
    using (var sr = new StringReader (xml))
        return Load (new XmlTextReader (sr) {
            Namespaces = false;
        });
}
```

Reply:
I use `Parse` and `ParseXml` to roughly match `XmlDocument.Load` and `XmlDocument.LoadXml` for the reason you state: `Parse/Load (string)` generally refers to a file rather than an XML snippet.

But `Load` and `Parse` work as well.